proposals/Fleetctl-trigger.md

```diff
@@ -0,0 +1,95 @@
+# Fleetctl trigger
+
+## Goal
+
+As a user, I would like to trigger a set of async jobs using `fleetctl`. For example, I’d like to
+trigger a vuln scan, or an MDM dep sync. 
+
+The proposed solution to accomplish this goal enables a new CLI command: 
+`fleetctl trigger --name <NAME>`.   
+
+## Background
+
+Currently, the Fleet server uses the `schedule` package to create sets of defined jobs that are run
+serially at defined intervals. The initial schedule interval must be specified at the point the
+schedule is instantiated via `schedule.New`. Optionally, `schedule. WithConfigReloadInterval`
+accepts a reload interval function. If specified, the reload interval function is periodically
+called and its return value becomes the new the schedule interval. This mechanism allows the
+schedule interval to be modified by user, for example, by changing the app config; however, there no
+mechanism currently to trigger async jobs on an ad hoc basis.
+
+## Proposal
+
+### New CLI command `fleetctl trigger --name <NAME>`
+- Upon this command, the CLI client makes a request to a new authenticated endpoint (see below) to
+  trigger an ad hoc run of the named schedule. 
+
+### New `schedule` option `WithTrigger` 
+- This option adds a `trigger` channel on the `schedule` struct that will trigger an ad hoc run of
+  the scheduled jobs.
+- The trigger channel for each `schedule` is exposed via a new `schedules` map on the `Service` struct. 
+
+### New endpoint `GET /trigger?name={:name}` 
+- The request handler first calls `ds.Lock` to check if the named schedule is locked.
```

Comment:
In my opinion, It would be better for the handler to just try to trigger the schedule, and have the schedule return an appropriate error like "already running error". Otherwise, the service has to know about locks, and the name of the lock, and that schedules use locks ... breaks the abstraction a bit.

Reply:
+1.

The manual trigger would just run the schedule just as if the ticker ticked. And return an "already running" error if `ds.Lock` returned `false, nil`.